dts: arm64: imx95: add TPM device nodes

Zhiqiang-Hou opened the pull request:
Added TPM device tree nodes for i.MX95 CA55.

Zhiqiang-Hou committed:
dts: arm64: imx95: add TPM device nodes

Added TPM device tree nodes for i.MX95 CA55.

Signed-off-by: Hou Zhiqiang <Zhiqiang.Hou@nxp.com>

JiafeiPan commented on dts/arm64/nxp/nxp_mimx95_a55.dtsi:
```diff
@@ -239,4 +239,70 @@
 		status = "disabled";
 	};
+
+	tpm1: tpm@44310000 {
```
would you please follow the address sequence to align the nodes? thanks.

Zhiqiang-Hou replied:
It's more readable putting the same IP block nodes together. In Linux it ordered the nodes by the unit address giving a bad experience.

JiafeiPan replied:
okay.

JiafeiPan commented:
Please also do the following updates:
1. upstream related hal patches
2. add overlay in tests/drivers/counter/counter_basic_api
3. update board document to add TPM support,

Thanks.

Zhiqiang-Hou commented:
> Please also do the following updates:
> 
> 1. upstream related hal patches
> 2. add overlay in tests/drivers/counter/counter_basic_api
> 3. update board document to add TPM support,
> 
> Thanks.

sure, thanks!

Zhiqiang-Hou committed:
boards: imx95: add TPM support in CA55 feature table

Add TPM counter feature in i.MX95 CA55 doc.

Signed-off-by: Hou Zhiqiang <Zhiqiang.Hou@nxp.com>

Zhiqiang-Hou committed:
tests: counter_basic_api: add imx95 evk support

Add i.MX 95 EVK support for counter_basic_api test application.

Signed-off-by: Hou Zhiqiang <Zhiqiang.Hou@nxp.com>

JiafeiPan reviewed (changes requested):
Please also update imx95_evk_mimx9596_a55.yaml to add counter in supports list, thanks.

Zhiqiang-Hou committed:
boards: imx95: add counter to support list of CA55 yaml

Update the yaml files of i.MX 95 CA55 and SMP CA55 variant to add
TPM counter to support list.

Signed-off-by: Hou Zhiqiang <Zhiqiang.Hou@nxp.com>

JiafeiPan commented:
Please rebase and trigger twister again as the building issue has been fixed by https://github.com/zephyrproject-rtos/zephyr/pull/81803

JiafeiPan commented on boards/nxp/imx95_evk/imx95_evk_mimx9596_a55.yaml:
```diff
@@ -14,4 +14,5 @@ toolchain:
 ram: 1024
 supported:
   - uart
+  - tpm counter
```
"counter" but not "tmp counter"

Zhiqiang-Hou replied:
fixed

JiafeiPan commented on boards/nxp/imx95_evk/imx95_evk_mimx9596_a55_smp.yaml:
```diff
@@ -15,4 +15,5 @@ ram: 1024
 supported:
   - smp
   - uart
+  - tpm counter
```
"counter" but not "tmp counter"

Zhiqiang-Hou replied:
fixed. Thanks!

JiafeiPan commented:
@Zhiqiang-Hou, this PR depend on hal update: https://github.com/zephyrproject-rtos/hal_nxp/pull/459, so please update west.yml to let twister build pass.

Zhiqiang-Hou committed:
west.yml: update hal_nxp reversion

Update the hal_nxp to the depended pull request.

Signed-off-by: Hou Zhiqiang <Zhiqiang.Hou@nxp.com>

Zhiqiang-Hou commented:
> @Zhiqiang-Hou, this PR depend on hal update: [zephyrproject-rtos/hal_nxp#459](https://github.com/zephyrproject-rtos/hal_nxp/pull/459), so please update west.yml to let twister build pass.

updated, thanks!

mmahadevan108 reviewed (approved):
Needs to be sequenced behind https://github.com/zephyrproject-rtos/zephyr/pull/79532
@Zhiqiang-Hou  I merged the HAL side. After the gating PR merges, then we can update the HAL SHA and I'll clear my CR

mmahadevan108 reviewed (changes requested):
Needs to be sequenced behind https://github.com/zephyrproject-rtos/zephyr/pull/79532
@Zhiqiang-Hou I merged the HAL side. After the gating PR merges, then we can update the HAL SHA and I'll clear my CR
